services/multitenant/tenantstats.go

```diff
@@ -241,6 +241,7 @@ func (t *Stats) GetRouterPickupJobs(destType string, noOfWorkers int, routerTime
 					timeRequired = 0
 				}
 				runningTimeCounter = runningTimeCounter - timeRequired
+				workspacePickUpCount[workspaceKey] = misc.MinInt(workspacePickUpCount[workspaceKey], runningJobCount)
```

Comment:
This was required to enforce the jobQueryBatchSize  limit in case a single workspace exists.
Overall, I think we are missing concise test scenarios and proper coverage for this critical piece of code